sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/azure_configuration_client_async.py

```diff
@@ -324,18 +456,72 @@ def list_configuration_settings(
             )
         """
         custom_headers = CaseInsensitiveDict(kwargs.get("headers"))
-        if_match = quote_etag(etag) if etag else None
-        key_value_deleted = await self._impl.delete_key_value(
-            key=key,
-            label=label,
-            if_match=if_match,
-            headers=custom_headers,
-            error_map={
-                409: ResourceReadOnlyError,
-                412: ResourceModifiedError,
-            },
-        )
-        return ConfigurationSetting._from_key_value(key_value_deleted)
+        if match_condition == MatchConditions.Unconditionally:
+            key_value_deleted = await self._impl.delete_key_value(
+                key=key,
+                label=label,
+                headers=custom_headers,
+                error_map={
+                    409: ResourceReadOnlyError,
+                },
```

Comment:
Here the calls all define `error_map` at the call site. Above, `error_map` is always defined before the call. Suggest being consistent throughout a given module.